AwsEncryptionSDK/runtimes/net/Examples/Keyring/AwsKmsHierarchical/AwsKmsHierarchicalKeyring.cs

```diff
@@ -47,162 +54,147 @@
+        var decryptInput = new DecryptInput
         {
-            KeyStore = keystore,
-            BranchKeyId = branchKeyIdB,
-            Cache = new CacheType { Default = new DefaultCache{EntryCapacity = 100} },
-            TtlSeconds = 600
+            Ciphertext = encryptOutput.Ciphertext,
+            EncryptionContext = encryptionContext,
+            Keyring = keyring
         };
-        var hierarchicalKeyringB = materialProviders.CreateAwsKmsHierarchicalKeyring(createHierarchicalKeyringB);
 
-        var decryptFailed = false;
-        try
-        {
-            // Try to use keyring for Tenant B to decrypt a message encrypted with Tenant A's key
-            // Expected to fail.
-            var decryptInput = new DecryptInput
-            {
-                Ciphertext = encryptOutputA.Ciphertext,
-                Keyring = hierarchicalKeyringB,
-            };
-            encryptionSDK.Decrypt(decryptInput);
-        }
-        catch (AwsCryptographicMaterialProvidersException)
-        {
-            decryptFailed = true;
-        }
-        Assert.True(decryptFailed);
+        // 7. Decrypt the Data
+        //    To decrypt data using the Hierarchical Keyring you need:
+        //    - "dynamodb:GetItem"
+        //    - "kms:Decrypt"
+        var decryptOutput = encryptionSdk.Decrypt(decryptInput);
         
-        decryptFailed = false;
-        try
-        {
-            // Try to use keyring for Tenant A to decrypt a message encrypted with Tenant B's key
-            // Expected to fail.
-            var decryptInput = new DecryptInput
-            {
-                Ciphertext = encryptOutputB.Ciphertext,
-                Keyring = hierarchicalKeyringA,
-            };
-            encryptionSDK.Decrypt(decryptInput);
-        }
-        catch (AwsCryptographicMaterialProvidersException)
-        {
-            decryptFailed = true;
-        }
-        Assert.True(decryptFailed);
-        
-        // Decrypt your encrypted data using the same keyring you used on encrypt.
-        encryptionSDK.Decrypt(new DecryptInput {
-            Ciphertext = encryptOutputA.Ciphertext,
-            Keyring = keyring }
-        );
-        encryptionSDK.Decrypt(new DecryptInput {
-            Ciphertext = encryptOutputB.Ciphertext,
-            Keyring = keyring }
-        );
-
+        // Demonstrate that the decrypted ciphertext and plaintext are the same
+        Assert.Equal(decryptOutput.Plaintext.ToArray(), plaintext.ToArray());
+        
+        // 8. Rotate the Branch Key in our KeyStore.
+        //    Only the branch key will be rotated. 
+        //    This rotation is eventually consistent. See the VersionBranchKey
+        //    Example for how to rotate a branch key.
+        //    To rotate a branch key you need:
+        //    - "dynamodb:GetItem"
+        //    - "kms:ReEncrypt*"
+        //    - "kms:GenerateDataKeyWithoutPlaintext"
```

Comment:
Suggestion: Link to our docs.
The VersionBranchKey file and these comments do not explain 
what rotation or an active branch key is.
We could add all that content here or in the other file.

Or we could link to our docs.
```suggestion
        // 8. Rotate the Branch Key in our KeyStore.
        //    Only the branch key will be rotated. 
        //    This rotation is eventually consistent. 
        //    See the VersionBranchKey for a narrower example.
        //    See the Developer Guide for details: https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/use-hierarchical-keyring.html#rotate-branch-key
        //    Example for how to rotate a branch key.
        //    To rotate a branch key you need:
        //    - "dynamodb:GetItem"
        //    - "kms:ReEncrypt*"
        //    - "kms:GenerateDataKeyWithoutPlaintext"
```